### PR TITLE
Delete cluster after test timeout.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
So far Jenkins kills the system test job after two hours. This also
kills our cleanup call. This patch introduces a bash timeout that still
leaves time for a cluster cleanup.

JIRA issues: MARATHON-8364

### CODE REVIEW

File: ci/si_install_deps.sh [hunk 1]
```diff
@@ -22,5 +22,10 @@ if ! command -v amm >/dev/null 2>&1; then
     echo "Ammonite successfully installed"
 fi
 
+# Ensure timeout is available.
+if [ "$PLATFORM" == 'Darwin' ]; then
```
Comment (zen-dog) [inline ci/si_install_deps.sh]: One has to have `brew install coreutils` but ok.

File: ci/si_pipeline.sh [hunk 2]
```diff
@@ -76,7 +76,7 @@ case $CLUSTER_LAUNCH_CODE in
   0)
       "$ROOT_PATH/ci/dataDogClient.sc" "marathon.build.si.$VARIANT.cluster_launch.success" 1
       cp -f "$DOT_SHAKEDOWN" "$HOME/.shakedown"
-      make test
+      timeout --preserve-status -s KILL 1s make test
```
Comment (zen-dog) [inline ci/si_pipeline.sh]: so we're killing after `1s`? 🤣 I don't think we're that fast yet

**Comment by jeschkies** [comment]
Validate in build https://jenkins.mesosphere.com/service/jenkins/job/system-integration-tests/job/marathon-si-pr/237/. 

1. The cluster is launched.
2. The test is killed after the timeout and exits with 137.
3. The cluster is deleted.

File: ci/si_pipeline.sh (hunk 2 above)
Comment (jeschkies) [reply ci/si_pipeline.sh]: We are getting there...

File: ci/si_install_deps.sh (hunk 1 above)
Comment (jeschkies) [reply ci/si_install_deps.sh]: I'll add that.

**Comment by jeschkies** [comment]
System test failed 🤔. We should investigate or restart.

**Pull request merged**